TGC 과제 - Todo list

HY0118 opened the pull request:
TGC 과제 - todo list 제작.

1. todo list 완료 여부를 CSS로 표현.
2. todo list 생성/삭제 버튼 추가.
3. todo list 개수 동적 표시.
4. todo list 생성/삭제에 대한 motion 추가.
5. title과 favicon 변경.

![image](https://github.com/creativeYS/midasit_tgc_web_dev/assets/40168374/d9f12952-f451-4a1e-8b87-2e8f7d61b20c)

creativeYS committed:
practice02

creativeYS committed:
practice02

creativeYS committed:
practice02
- Component, Hooks example
- update readme.md

HY0118 committed:
Merge remote-tracking branch 'upstream/main' into lhy0118

HY0118 committed:
TGC 과제 - todo list 제작

1. todo list 완료 여부를 CSS로 표현.
2. todo list 생성/삭제 버튼 추가.
3. todo list 개수 동적 표시.
4. todo list 생성/삭제에 대한 motion 추가.
5. title과 favicon 변경.

creativeYS commented on practice02/src/components/Todos.css:
```diff
@@ -0,0 +1,19 @@
+}
+
+#line_complete {
+  text-decoration: red line-through;
```
깔끔하게 붉은 줄 그어주셔서 보기 좋네요!

creativeYS commented on practice02/src/components/Todos.js:
```diff
@@ -0,0 +1,132 @@
+          alignItems: "center",
+          width: "670px",
+          marginLeft: "20px",
+        }}
```
inline 으로 개별 html 요소에 style을 적용하기보다는 class 를 사용하여 css 파일로 분리하는 것이 소스내에서 html 구조 파악과 간결하게 보여주는데 도움이 됩니다!

```
// Todos.js
<div className="TodosHeader" ...>

// Todos.css
.TodosHeader {
  display: flex;
  ...
}
```

creativeYS commented on practice02/src/components/Todos.js:
```diff
@@ -0,0 +1,132 @@
+            <FaTrash />
+          </button>
+        </div>
+      </div>
```
하나의 컴포넌트 (Todos) 안에 내용이 많아서 분리를 시도해보는 것도 좋겠습니다.
![image](https://github.com/creativeYS/midasit_tgc_web_dev/assets/37997667/6faa05e2-6328-4d73-bd29-38937a8fd898)
두 부분으로 쪼갤 수 있을 것 같아요!

컴포넌트를 다음과 같이 새로 만든다면 Todos 컴포넌트에서 정의한 상태 counter를 전달받아서 쓸 수 있을 것 같습니다.

```
// TodoHeader.js
import React from "react";

function TodoHeader({counter, setCounter}) {
    return (
        <div>
        </div>
    );
}
export default TodoHeader;

//Todo.js
  ...
  return (
    <div className="Todos">
      <TodoHeader counter={counter} setCounter={setCounter}/>
      ...
    </div>
  )
```

[추가 정보]
1. 현재는 상태 관리 도구를 붙이지 않아서 상태를 드릴링해야 하는데요, 프로젝트 규모가 크지 않기 때문에 현재는 크게 문제 없을 것 같습니다.

creativeYS commented on practice02/src/components/Todos.js:
```diff
@@ -0,0 +1,132 @@
+                  borderBottom: "1px solid #000",
+                }}
+                id={
+                  todo.completed === true ? "line_complete" : "line_incomplete"
```
id 값을 가지고 css에서 완료 여부를 표현하기 보다는
className으로 표현하는게 좋습니다. 왜냐하면 id는 문서에서 유일해야 하기 때문입니다. 따라서 본 js 코드와 css 코드를 수정해보아요~
```
<motion.li ...
  className={todo.completed === true ? "line_complete" : "line_incomplete"} ...>
```

creativeYS commented:
역시...! 결과도 이쁘고 코드도 잘 작성되었습니다.
동작에 전혀 문제는 없지만 좀더 심화를 위하여 몇가지 피드백을 작성하였습니다. 확인한번 해주시면 감사하겠습니다.
(다른 구성원들도 참고하면 좋을 것 같아서 좀 자세하게 적었사오니 잘 부탁드립니다. 하영님. ㅎㅎㅎ 😊)

HY0118 committed:
Code Review 반영

js에서 스타일 코드 css로 분리.
기능 별 컴포넌트 분리 (Liststate, Listbutton).
id 기준 css 변경 조건식을 className으로 변경.

HY0118 commented:
@creativeYS 
코드 리뷰 해주신 사항에 대해 수정해서 커밋하였습니다~! 꼼꼼한 리뷰 감사합니다 👍

creativeYS commented:
멋져요!
코드 병합하겠스빈다!
